apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -604,6 +603,27 @@ const allFieldsHidden = (fieldVisibilities: AichatState['fieldVisibilities']) =>
     key => fieldVisibilities[key] === Visibility.HIDDEN
   );
 
+const updateMessagesOnError = (
+  newMessage: ChatCompletionMessage,
+  dispatch: ThunkDispatch<unknown, unknown, AnyAction>
+) => {
+  const assistantChatMessage: ChatCompletionMessage = {
+    id: getNewMessageId(),
+    role: Role.ASSISTANT,
+    status: Status.ERROR,
+    chatMessageText: 'error',
+    timestamp: getCurrentTimestamp(),
+  };
+  dispatch(addChatMessage(assistantChatMessage));
+
+  dispatch(
+    updateUserChatMessageStatus({
+      id: newMessage.id,
+      status: Status.ERROR,
```

Comment:
It makes sense to me to keep the user message that resulted in an error response in view, so the user gets some feedback that they did indeed attempt to submit a message and we had an issue processing it. I could imagine it being confusing if your message just disappeared...what do you think?

Worth noting that we do exclude these "unfulfilled" user messages from future conversation history that is submitted to the model here:

https://github.com/code-dot-org/code-dot-org/blob/ea984eaabd11973b70e13530e7e30d4c2246673b/dashboard/app/controllers/aichat_controller.rb#L44

Reply:
Thanks! and yes, I agree it would be confusing if your message just disappeared.

I wonder if it would make sense to indicate via UI that this user message is in an error state to reflect that it's not part of the chat conversation with the chatbot. This is outside the scope of this PR but if the chat workspace is acting as a kind of history for the user, it could be useful to distinguish an errored/unfulfilled user message and an okay/fulfilled one.